applications: Added DFU over SMP support to the Matter Bridge (#12367)
nrfconnect/sdk-nrf: applications/matter_bridge/README.rst

* Temperature Sensor
* Humidity Sensor

The application supports over-the-air (OTA) device firmware upgrade (DFU) using one of the two following protocols:

> * Matter OTA update protocol that uses the Matter operational network for querying and downloading a new firmware image.

Reviewer: ```suggestion
* Matter OTA update protocol.
  Querying and downloading a new firmware image is done using the Matter operational network.
```
Wanted to break this into two sentences so the protocol is its own sentence, which meant rearranging the rest of it. Make sure that the meaning is still the same.

* Simple Management Protocol (SMP) over Bluetooth LE.

>   In this case, the DFU can be done either using a smartphone application or a PC command line tool.

Reviewer: ```suggestion
  The DFU is done using either a smartphone application or a PC command line tool.
```

  Note that this protocol is not part of the Matter specification.

In both cases, MCUboot secure bootloader is used to apply the new firmware image.

> The DFU over Matter is enabled by default.

Reviewer: ```suggestion
Matter OTA update protocol is enabled by default.
```

> To configure the sample to support the DFU over Matter and SMP, use the ``-DCONFIG_CHIP_DFU_OVER_BT_SMP=y`` build flag.

Reviewer: ```suggestion
To configure the application to support the DFU over both Matter and SMP, use the ``-DCONFIG_CHIP_DFU_OVER_BT_SMP=y`` build flag.
```

See :ref:`cmake_options` for instructions on how to add these options to your build.

> When building on the command line, run the following command with *build_target* replaced with the build target name of the hardware platform you are using (see `Requirements`_), and *dfu_build_flag* replaced with the desired DFU build flag:
> 
> .. parsed-literal::
>    :class: highlight
> 
>    west build -b *build_target* -- *dfu_build_flag*
> 
> For example:

Reviewer: ```suggestion
When building on the command line, run the following command:

.. parsed-literal::
   :class: highlight

   west build -b *build_target* -- *dfu_build_flag*

Replace *build_target* with the build target name of the hardware platform you are using (see `Requirements`_), and *dfu_build_flag* with the desired DFU build flag.
For example:
```
I think this flows a little better.

.. code-block:: console

   west build -b nrf7002dk_nrf5340_cpuapp -- -DCONFIG_CHIP_DFU_OVER_BT_SMP=y

> For information about how to upgrade the device firmware using a PC or a mobile, see the :ref:`matter_bridge_app_dfu` section.

Reviewer: ```suggestion
For information about how to upgrade the device firmware using a PC or a smartphone, see the :ref:`matter_bridge_app_dfu` section.
```
We called it a smartphone earlier in the section, so using the same word here.

    :end-before: matter_door_lock_sample_led1_end

Button 1:
    Depending on how long you press the button:

>     * If pressed for less than three seconds, it initiates the SMP server (Security Manager Protocol).

Reviewer: ```suggestion
    * If pressed for less than three seconds, it initiates the SMP (Security Manager Protocol) server.
```
Different SMP? It was Simple Management Protocol earlier.

Author: Wow I haven't noticed it before. Of course you are right. I basically did a copy-paste from existing docs in another sample, so it looks like we have this issue from a while. I will align another samples as well.

      After that, the Direct Firmware Update (DFU) over Bluetooth Low Energy can be started.
      (See `Upgrading the device firmware`_.)
    * If pressed for more than three seconds, it initiates the factory reset of the device.

>       Releasing the button within the 3-second window cancels the factory reset procedure.

Reviewer: Is it really correct calling it canceling the reset if it's starting the SMP server? Or am I misunderstanding the 3-second window?

Author: Let me explain it then, it goes like this:
0 s < press time < 3 s - enables DFU over SMP
3 s <= press time < 6 s - initiation of factory reset (basically it's only about printing a warning that factory reset is going to happen soon, so user can react and release the button to cancel it)
press time = 6 s - actual factory reset is done

    :end-before: matter_door_lock_sample_led1_end

Button 1:
    Depending on how long you press the button:

    * If pressed for less than three seconds, it initiates the SMP server (Simple Management Protocol).
      After that, the Direct Firmware Update (DFU) over Bluetooth Low Energy can be started.
      (See `Updating the device firmware`_.)

>     * If pressed for more than three seconds, it initiates the factory reset of the device.
>       Releasing the button within the 3-second window cancels the factory reset procedure.

Reviewer: Ah, that clarifies it.
```suggestion
    * If pressed for more than three seconds, it initiates the factory reset of the device.
      Releasing the button within a 3-second window of the initiation cancels the factory reset procedure.
```
